### ISSUE TITLE
continuity policy and package registries?

### ISSUE THREAD

**Issue by ctcpip** [openjs-foundation/cross-project-council#1355]
it didn't occur to me before when discussing #1327, but do we also need to address this issue wrt package registries, e.g. `npm` ?

cc @bensternthal

**Comment by bensternthal** [comment]
@ctcpip I 100% agree. We can tackle this next.

**Comment by mcollina** [comment]
node.js has https://www.npmjs.com/~nodejs-foundation

**Comment by tobie** [comment]
Discussed in the CPC call today. General agreement to include this in the continuity policy. @bensternthal to open a PR.

**Comment by tobie** [comment]
**Notes from this week's CPC call:**

- Using HospitalRun as a test case
- Issue transferring the package
- Maintainers filed issues with NPM
- Waiting on the result of having filed this issue
- Discussion in CPC suggests we should bring this to the security collab space

**Comment by 0xAverageUser** [comment]
@bensternthal @ctcpip, dropping my thoughts here after today's meeting.

I think the issue spans a spectrum based on the number of maintainers per project. The policy result would be very different depending on that number. 

- For projects with many maintainers (think 20+ on npm): Adding an OpenJS maintenance account creates a honeypot, increases risk with a single point of failure.
- Few or no maintainers (0-3 active): OpenJS plays a critical role in supporting, takeover, archiving, and emergency admin removal (e.g., xz-utils).

These two extremes are I think orthogonal. The policy should adapt to the number of maintainers. If this is relevant, I'm happy to contribute.

**Comment by ruddermann** [comment]
This was discussed in the 11 March CPC Working Session (attended by @bensternthal @joesepi @kj-powell @ruddermann). 

**NOTE: The next meeting on this topic will involve reviewing the initial draft npm Continuity Policy in this Monday 17 March's Security Collab Space. Please attend if you'd like to contribute!**

After reviewing [npm Org & Account Management and Permissions Overview DRAFT](https://docs.google.com/document/d/1iaNEO8pYjkeog_rqASWaa9rjn_8kHpCWDXa7qy2G418/edit?tab=t.0) we decided:

- Create a new npm Continuity Policy doc with an approach that involves adding an OpenJS account to Maintainer managed npm Organization(s) inspired by the [OpenJS GitHub Continuity Policy](https://github.com/openjs-foundation/cross-project-council/blob/main/project-resources/github-continuity-policy.md).
- OpenJS will create a unique user account for each npm Organization.
- The credentials and 2FA for each account will be stored in a separate 1Password vault only accessible by 1-2 key OpenJS staff (likely @rginn and @bensternthal).
- Include compliance to this policy in the [OpenJS project onboarding checklist](https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_PROGRESSION.md)
- Include best practices and recommendations related to npm in the Q1'25 update to the [Security Compliance Guidelines](https://docs.google.com/spreadsheets/d/1GwIsAudAn89xv9DAbr1HUaY4KEVBsYfg--_1cW0uIB0/edit?pli=1&gid=791581093#gid=791581093).

**Next Steps**

[ ] Proceeding with drafting a new OpenJS npm Continuity Policy this week. This will be done by @ruddermann in the Continuity Policy tab of the [DRAFT npm gDoc](https://docs.google.com/document/d/1iaNEO8pYjkeog_rqASWaa9rjn_8kHpCWDXa7qy2G418/edit?tab=t.0) .
[ ] Add review of the new draft policy to 17 March Security Collab Space for discussion and feedback

**Comment by tobie** [comment]
**Notes from this week's CPC call:**
 
- Security collab space working on this
- Will finalize the doc and bring it to the CPC for approval

**Comment by wesleytodd** [comment]
> I'm not entirely clear on this aspect. Do you mean that a new user will be created, for example, one for Node.js, another for Express, and so on for each project?

Yes I believe this is the correct understanding. The goal here is that compromising one org would not impact others IIUC.

**Comment by bensternthal** [comment]
Waiting on written recommendation from @ruddermann

**Comment by ruddermann** [comment]
Howdy folks, apologies for the delay replying here!

I've written a draft of the policy, which you can find in [this Google Doc](https://docs.google.com/document/d/1iaNEO8pYjkeog_rqASWaa9rjn_8kHpCWDXa7qy2G418/edit?tab=t.j66rkgmy0xcr#heading=h.2j71n6oo9o30). I'll include it below for easy viewing for folks. Feel free to comment here or in the gDoc!

# DRAFT OpenJS npm Continuity Policy

***NOTES:***

- *Document structure and policy language are borrowed from the [OpenJS GitHub Continuity Policy](https://github.com/openjs-foundation/cross-project-council/blob/main/project-resources/github-continuity-policy.md).*  
- *Expect additional Continuity Policies for other package registries in the future.*

---

## Summary

This policy is intended to ensure that OpenJS npm Organizations remain accessible and manageable.

It addresses the following use cases:

* Managing projects and users when maintainers are inactive or unreachable  
* [Deprecating packages](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions) when maintainers are no longer reachable  
* Adding or removing npm Organization members in emergency situations

## Continuity Options for Projects

**Option 1: Add an OpenJS Continuity Account to your npm Organization(s) and/or Packages**

If you select this option, a new and unique OpenJS account managed by the Executive Director (currently [https://github.com/rginn](https://github.com/rginn)) will be created for your Project. This can be done by creating an issue in the [Infrastructure repository](https://github.com/openjs-foundation/infrastructure/issues/new?template=IT-SUPPORT-REQUEST.yml).

* If you use npm Organizations, you will add this account to your Organization(s) and grant it the Owner role.  
* If you don’t use npm Organizations, you will add this account as a Maintainer to your npm Packages.

**Option 2: Opt Out**

Projects that feel their governance is sufficient to provide continuity may opt out of this policy by requesting an exception from the CPC. This can be done by filing an issue in the [CPC repository](https://github.com/openjs-foundation/cross-project-council/issues).

- Organizations with 1-2 maintainers in the Owner role are the highest priority for this policy and are unlikely to be granted an exception.  
- Organizations with 3 or more maintainers in the Owner role who have been active in the past \[6||12\] months will be considered for an exception.

**Comment by ruddermann** [comment]
Regarding the npm continuity policy, I've also updated the [this Google Doc](https://docs.google.com/document/d/1iaNEO8pYjkeog_rqASWaa9rjn_8kHpCWDXa7qy2G418/edit?tab=t.j66rkgmy0xcr#heading=h.2j71n6oo9o30) to incorporate feedback from @bensternthal.

I didn't see an agenda issue for the CPC meeting tomorrow, but this issue should still be tagged correctly to put it on the agenda.

**Comment by tobie** [comment]
**Notes from this week's CPC call:**

* [New draft](https://docs.google.com/document/d/1iaNEO8pYjkeog_rqASWaa9rjn_8kHpCWDXa7qy2G418/edit?tab=t.j66rkgmy0xcr) from @ruddermann. Will be turned into PR asap.

**Comment by mgol** [comment]
The jQuery project added the [`openjsfoundation`](https://www.npmjs.com/~openjsfoundation) npm user to its packages a while ago; example: [the `jquery` package](https://www.npmjs.com/package/jquery). I assume this means we'll need to request a new user, add it to all packages as a maintainer (we don't use scoped packages for historical reasons) and then remove the `openjsfoundation` user from the lists?

**Comment by mgol** [comment]
Also, one question: some projects may miss these new requirements. How do we ensure all projects will follow the procedure?

**Comment by tobie** [comment]
Hey @bjohansebas, did you close this on purpose or accidentally?

**Comment by bjohansebas** [comment]
https://github.com/openjs-foundation/cross-project-council/pull/1483 was merged, are there any other steps to follow? Apart from knowing what the projects will choose that’s being tracked in https://github.com/openjs-foundation/command-center/pull/70 and https://github.com/openjs-foundation/cross-project-council/issues/1486.

Feel free to reopen if there’s still work to do here

**Comment by tobie** [comment]
OK, I was missing the rationale for closing. Thanks for providing it.
